### ISSUE TITLE
query with .include(key) sometimes return key with full data sometimes as pointer only

### ISSUE THREAD

**Issue by refre5h** [ParsePlatform/parse-server#609]
We have functions like this in our cloud code:
```
Parse.Cloud.define("findComments", function(request, response) {
    var query = new Parse.Query("Comment");
    query.include("post"); // "post" column is a pointer to another class Post
    query.equalTo("isExpired",false);
    query.find().then(function(results){
        res.response(results)
    }, function(error){
        res.error(error);
    })
})
```

The `results` look like below. Notice the first result has post as `"__type": "Pointer"` (which causes the client to crash as it expects value from this key), whereas the second has it as `"__type": "Object"` that includes all info the client needs.
```
[{
    "isExpired": false,
    "post": {
        "__type": "Pointer",
        "className": "Post",
        "objectId": "5eW6o54GHI"
    },
    "updatedAt": "2016-02-23T02:21:52.722Z",
    "createdAt": "2016-02-23T02:21:52.722Z",
    "objectId": "PN0qx4Qqi0"
},{
    "isExpired": false,
    "post": {
        "creator": {
            "__type": "Pointer",
            "className": "_User",
            "objectId": "gTGJZHahwA"
        },
        "updatedAt": "2016-02-22T20:47:35.761Z",
        "createdAt": "2016-02-22T20:47:35.761Z",
        "objectId": "ndb24UejOF",
        "__type": "Object",
        "className": "Post"
    },
    "updatedAt": "2016-02-22T20:48:52.924Z",
    "createdAt": "2016-02-22T20:47:35.903Z",
    "objectId": "T2K0FNK1PA"
}]
```

**Comment by aziz92** [comment]
Does the object Post with id 5eW6o54GHI exist?

**Comment by refre5h** [comment]
@aziz92 Yes it exists. I questioned the same thing while we discovered this.
```
{
    "_id": "5eW6o54GHI",
    "_p_creator": "_User$gTGJZHahwA",
    "_updated_at": {
        "$date": "2016-02-23T02:21:52.289Z"
    },
    "_created_at": {
        "$date": "2016-02-23T02:21:52.289Z"
    }
}
```

**Comment by flovilmart** [comment]
This should be fixed on 2.1.4, I'll reopen if you still have issues
